### ISSUE TITLE
When Copy is Performed, Internet Explorer 11 Scrolls Webpage Almost to the Top

### ISSUE THREAD

**Issue by br-2000** [omichelsen/angular-clipboard#48]
The problem occurs on IE 11.0.33 on Windows 7.

Whenever copyText() is called, IE 11 scrolls a long page almost to the top.  Thus, if a copy operation is performed somewhere in the middle or bottom of a page, a user's viewport suddenly jumps to the top.

I was able to reproduce the problem easily using your own demo.html. To reproduce take demo.html and insert:
`
<p> line 1 </p>
<p> line 2 </p>
...
<p> line 99 </p>
<p> line 100 </p>
`
just before h1: "Click button to copy text to clipboard"

Then click on "Copy" button, and IE would scroll the page up.

I attach modified demo.html (renamed to ie11_scroll.txt because of restriction on attaching html files).
[ie11_scroll.txt](https://github.com/omichelsen/angular-clipboard/files/426196/ie11_scroll.txt)

**Comment by br-2000** [comment]
After some investigation and checking the source code of clipboard.js, I found that adding $window dependency and modifying createNode() as follows:
 ```
if (context instanceof HTMLElement) {
  // commented out:
  // node.style.top = context.getBoundingClientRect().top + 'px';
}
// added:
node.style.top = ($window.pageYOffset || $document.documentElement.scrollTop) + 'px';
```
solves the problem with random scroll to the top in IE.

Btw, while looking at clipboard.js source, I found other special case handling in the code, e.g., prevent zooming in iOS, RTL, etc.

**Comment by ipundit** [comment]
Confirmed the bug on IE 11.545.10586.0 and Edge 25.10586.0.0 on Windows 10.  Firefox and Chrome are fine.  br-2000's fix works IE and Edge, without changing the behavior for Firefox and Chrome on Windows 10.

This is a showstopper issue for me.  Can the fix be promoted?

**Comment by omichelsen** [comment]
I've added your suggested fix as v1.5.0.
